application-ckeditor-plugins/src/main/resources/xwiki-macro/macroEditor.js

```diff
@@ -57,10 +58,30 @@ define('macroService', ['jquery'], function($) {
       });
     }
     return deferred.promise();
+  },
+
+  installMacro = function(extensionId, extensionVersion) {
+    var deferred = $.Deferred();
+
+    var url = new XWiki.Document('MacroService', 'CKEditor').getURL('get', $.param({
+      outputSyntax: 'plain',
+      language: $('html').attr('lang')
+    }));
+    $.post(url, {action: 'install', extensionId: extensionId, extensionVersion: extensionVersion,
+        /*jshint camelcase: false */
+    	'form_token': xcontext.form_token})
+      .done(function(jobStatus) {
+        deferred.resolve(jobStatus);
+      }).fail(function() {
+        deferred.reject.apply(deferred, arguments);
+      });
```

Comment:
```suggestion
    return $.post(url, {action: 'install', extensionId: extensionId, extensionVersion: extensionVersion,
        /*jshint camelcase: false */
    	'form_token': xcontext.form_token});
```

You don't need a separate ``$.Deferred`` object, you can return directly the result of ``$.post()``. See https://api.jquery.com/jQuery.post/

> This jQuery XHR object, or "jqXHR," returned by $.post() implements the Promise interface, giving it all the properties, methods, and behavior of a Promise (see [Deferred object](https://api.jquery.com/category/deferred-object/) for more information).

Reply:
Right, I initially had different behaviors and I forgot to cleanup :)